### PR TITLE
Add support for excluding segments

### PR DESCRIPTION
**Pull request by julienrf**
Relates to scylladb/scylla-migrator#165.

### CODE REVIEW

File: emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/read/AbstractDynamoDBInputFormat.java [hunk 1]
```diff
@@ -65,13 +70,26 @@ public InputSplit[] getSplits(JobConf conf, int desiredSplits) throws IOExceptio
...
+    Set<Integer> excludedSegments =
+        Arrays.stream(conf.getInts(DynamoDBConstants.EXCLUDED_SCAN_SEGMENTS))
+            .boxed()
+            .collect(Collectors.toSet());
 
-    return getSplitGenerator().generateSplits(numMappers, numSegments, conf);
+    List<Integer> segments =
+        IntStream.range(0, numTotalSegments)
+            .boxed()
+            .filter(i -> !excludedSegments.contains(i))
```
Comment (tarzanek) [inline emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/read/AbstractDynamoDBInputFormat.java:84]: will this work even if there are overlaps or subsets?

in CQL you have token ranges, which are sized by split count, but algorithm counts with that


are we sure here that if e.g. input params (such as number of mappers or read throughput) change that this algorithm will still work?

or we mandate same input params?

**Comment by tarzanek** [comment]
so 
is that way how you generate segments idempotent and safe EVEN if input params change?

**Comment by julienrf** [comment]
I am assuming that the content of segments is deterministic for a given number of items, but I realize that I didn’t check that assumption. I’ve documented the requirements here:

https://github.com/scylladb/scylla-migrator/pull/190/files#diff-97b9fd1ccdf89c2312cb209e10885c6f18583ccaa8d9ddda7308401aa7a26628R15

In short, as long as the number of items in the table didn’t change, the segments map to the same items.

**Comment by tarzanek** [comment]
hmm, but tables are live, so number of items changes constantly

**Comment by julienrf** [comment]
According to [this](https://www.reddit.com/r/aws/comments/po6pmt/dynamodb_parallel_segment_scan_resulting_in_one/), the segment an item belongs to is stable even if more items are added to the table after or even during the migration.

So, the approach is reliable as long as AWS does not change the way scan segments work.

**Comment by guy9** [comment]
@tarzanek any other input on this?

**Pull request merged**